src/org/sosy_lab/java_smt/solvers/z3/Z3TheoremProver.java

```diff
@@ -169,8 +184,12 @@ public void close() {
 
       Native.solverReset(z3context, z3solver); // remove all assertions from the solver
       Native.solverDecRef(z3context, z3solver);
+      if (propagator != null) {
+        propagator.close();
```

Comment:
set propagator to null after cleanup? Otherwise a second call to "close" would result in a NullPointer.

Reply:
I set it to null now, but in practice it would not be an issue because all of this code can only be executed once due to the surrounding condition.